### PR TITLE
Converting src/browser-index.js to typescript

### PR DESCRIPTION
**Pull request by Aryaman1706**
Signed-off-by:  Aryaman Grover <1761ary@gmail.com>

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Comment by Aryaman1706** [comment]
Hello all!
I am having a look at failing tests and would fix soon.

**Comment by Aryaman1706** [comment]
@SimonBrandner Can you please have a look and let me know if anything else is required. 
Thank You.

**Review by SimonBrandner — COMMENTED** [review]
Thanks for working on this! 

I am not the most knowledgeable about this part of the app, so better to leave the review to someone else. A review from a team member is required anyway

**Comment by Aryaman1706** [comment]
@SimonBrandner Thank you for the response.
Can you let me know who all are the active maintainers so that I could tag them and ask for the review?

**Comment by SimonBrandner** [comment]
> Can you let me know who all are the active maintainers so that I could tag them and ask for the review?

A review from them has already been requested. You can see review requests in the top right corner

**Comment by Aryaman1706** [comment]
@dbkr Kindly have a look please.
Let me know if any further changes are required.

File: src/browser-index.ts [hunk 1]
```diff
@@ -16,9 +16,23 @@ limitations under the License.
...
 
-matrixcs.request(function(opts, fn) {
+interface RequestOptions {
+    qs: string | Record<string | number | symbol, string>;
+    qsStringifyOptions: IStringifyOptions;
+}
+
+enum IndexedDBStoreName {
+    crypto = "matrix-js-sdk:crypto"
+}
```
Comment (turt2live) [inline src/browser-index.ts:28]: This should be a constant

File: src/browser-index.ts [hunk 2]
```diff
@@ -16,9 +16,23 @@ limitations under the License.
...
+
+type GlobalObject = NodeJS.Global & typeof globalThis & Record<"matrixcs", typeof matrixcs>;
```
Comment (turt2live) [inline src/browser-index.ts:30]: I thought we already had a type for this? In general I don't think we should be touching `global` and should fix it.

File: src/browser-index.ts [hunk 3]
```diff
@@ -16,9 +16,23 @@ limitations under the License.
...
 import * as matrixcs from "./matrix";
 import request from "browser-request";
-import queryString from "qs";
+import queryString, { IStringifyOptions } from "qs";
 
-matrixcs.request(function(opts, fn) {
+interface RequestOptions {
+    qs: string | Record<string | number | symbol, string>;
+    qsStringifyOptions: IStringifyOptions;
+}
```
Comment (turt2live) [inline src/browser-index.ts:24]: we should be importing this type, not defining it.

**Review by turt2live — CHANGES_REQUESTED** [review]
Overall it seems fine - thanks! There's a few pieces to fix up here, and some merge conflicts to consider.

**Comment by t3chguy** [comment]
@Aryaman1706 are you able to continue with this PR?

**Comment by t3chguy** [comment]
This happened

**Pull request closed without merging**